### ISSUE TITLE
Support for OpenLava versions 2.2/3.x?

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#152]
OpenLava is an alternative open source implementation of Platform Load Facility Sharing (LSF) batch scheduler by IBM. 

OpenLava is almost compatibile with LSF however it does not recognise the `-cwd` option to set the job working director. For example: 

    #!/bin/bash 
    #BSUB -cwd /nfs/nextflow/work/4a/544a1111e5ba226fd60be0f9627282
    #BSUB -o /nfs/nextflow/work/4a/544a1111e5ba226fd60be0f9627282/.command.log
    #BSUB -J nf-startup_(1)
    [etc.]

**Comment by pditommaso** [comment]
I've uploaded a snapshot that do not set the `#BSUB -cwd` in the LSF job wrapper script. This should be enough to make it compatible with OpenLava. 

Could you test it? 

In order to download and run this snapshot, define this variable in your environment: 

    export NXF_VER=0.19.0-SNAPSHOT

Then run `nextflow info` to download it. It should print: 

     $ nextflow info
      Version: 0.19.0-SNAPSHOT build 3713
      Modified: 11-05-2016 08:30 UTC (10:30 CEST)

Then use nextflow as usual

**Comment by amacbride** [comment]
Yes, that seems to work just fine. Still doing some testing to see if I run into any other issues, but so far, so good!

**Comment by pditommaso** [comment]
Good. It would be great if you could also test jobs defining resource requirements such as `cpus`, `memory` and `time`. 

https://github.com/nextflow-io/nextflow/blob/master/src/main/groovy/nextflow/executor/LsfExecutor.groovy#L57-L81

**Comment by pditommaso** [comment]
@mfoll What about this change in the LSF executor (basically replacing `$BSUB -cwd /path` with a `cd /path` in the job wrapper)? Any possible side effects?

**Comment by mfoll** [comment]
I can't think of any side effect, but I'm not a real expert of LSF. I can try to run a few examples in my LSF cluster if you push a version like that.

**Comment by pditommaso** [comment]
I guess you have already used it!

**Comment by amacbride** [comment]
I'm seeing some intermittent problems with hanging processes, but a general problem I'm seeing is oddities in the work directory. I see directories that have a colon appended at the end, and a hierarchy mirroring the full path. It looks like it might be a bad Docker volume specifier?

Inside the work directory (/mnt/work), I'm seeing the directory I expect, plus this one:
```
160406_NB501331_0019_AH3W23BGXY/450QJ5BH/af/a6e363cbbdab6a39f0f252653b601e:
└── mnt
    └── work
        └── 160406_NB501331_0019_AH3W23BGXY
            └── 450QJ5BH
                └── af
                    └── a6e363cbbdab6a39f0f252653b601e
```

**Comment by amacbride** [comment]
The docker run command used the following spec (on a subsequent run):

```
-v /mnt/work/160406_NB501331_0019_AH3W23BGXY/450QJ5BH/56/96e1bb341287b364f5879f61d028c3:/mnt/work/160406_NB501331_0019_AH3W23BGXY/450QJ5BH/56/96e1bb341287b364f5879f61d028c3 -v /mnt/work/160406_NB501331_0019_AH3W23BGXY/450QJ5BH/56/96e1bb341287b364f5879f61d028c3:/mnt/work/160406_NB501331_0019_AH3W23BGXY/450QJ5BH/56/96e1bb341287b364f5879f61d028c3 -w /mnt/work/160406_NB501331_0019_AH3W23BGXY/450QJ5BH/56/96e1bb341287b364f5879f61d028c3
```

and when I look in those directories, I see:

```
ls /mnt/work/160406_NB501331_0019_AH3W23BGXY/450QJ5BH/*
/mnt/work/160406_NB501331_0019_AH3W23BGXY/450QJ5BH/56:
96e1bb341287b364f5879f61d028c3/  96e1bb341287b364f5879f61d028c3:/
```

**Comment by pditommaso** [comment]
Weird. Could you please attach the .command.run for that job?

p

**Comment by pditommaso** [comment]
@amacbride The support for OpenLava has been included in version `0.19.0`. If you are experiencing problems please open a new issue.
